src/main.cpp

```diff
@@ -1171,6 +1171,23 @@ bool CheckZerocoinSpend(const CTransaction& tx, bool fVerifySignature, CValidati
     return fValidated;
 }
 
+
+bool CheckColdStake(const CTransaction& tx, CValidationState& state)
+{
+    CTxOut prevOut;
+    if(!GetOutput(tx.vin[0].prevout.hash, tx.vin[0].prevout.n, state, prevOut))
+        return state.DoS(100, error("%s : invalid input", __func__), REJECT_INVALID, "bad-txns-inputs");
+
+    if (!prevOut.scriptPubKey.IsPayToColdStaking())
+    	return true;
+
+    // spending to the same contract
+   	if (prevOut.scriptPubKey != tx.vout[1].scriptPubKey)
+        return state.DoS(100, error("%s : invalid scripts", __func__), REJECT_INVALID, "bad-txns-cold-stake");
+
```

Comment:
This needs to check every vout from 1 to n-1 to be the same and not only the first one.

Reply:
this is already done when `OP_CHECKCOLDSTAKEVERIFY` is included.
So a TX not verifying that would fail at the script check (https://github.com/PIVX-Project/PIVX/pull/955/commits/8df2deb4ce4021e5cad9ed0cbb083f0016f9328b)